content/en/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md

```diff
@@ -1,76 +1,201 @@
 
-Through the Metrics API, you can get the amount of resource currently used
-by a given node or a given pod. This API doesn't store the metric values,
-so it's not possible, for example, to get the amount of resources used by a
-given node 10 minutes ago.
+{{< note >}}
+The resource metrics pipeline only supports CPU and memory metrics for use by HPA and VPA. To access a more complete set of metrics, you will need to deploy a [Full metrics pipeline](https://kubernetes.io/docs/tasks/debug-application-cluster/resource-usage-monitoring/#full-metrics-pipeline). 
+{{< /note >}}
 
-The API is no different from any other API:
 
-- it is discoverable through the same endpoint as the other Kubernetes APIs under the path: `/apis/metrics.k8s.io/`
-- it offers the same security, scalability, and reliability guarantees
+Figure 1 illustrates the architecture of the resource metrics pipeline.
+
+{{< mermaid >}}
+flowchart RL
+subgraph cluster[Cluster]
+direction RL
+S[ <br><br> ]
+A[Metrics-<br>Server]
+subgraph B[Nodes]
+direction TB
+D[cAdvisor] --> C[kubelet]
+E[Container<br>runtime] --> D
+E1[Container<br>runtime] --> D
+P[pod data] -.- C
+end
+L[API<br>server]
+W[HPA]
+C ---->|Summary<br>API| A -->|metrics<br>API| L --> W
+end
+L ---> K[kubectl<br>top]
+classDef box fill:#fff,stroke:#000,stroke-width:1px,color:#000;
+class W,B,P,K,cluster,D,E,E1 box
+classDef spacewhite fill:#ffffff,stroke:#fff,stroke-width:0px,color:#000
+class S spacewhite
+classDef k8s fill:#326ce5,stroke:#fff,stroke-width:1px,color:#fff;
+class A,L,C k8s
+{{< /mermaid >}}
+
+Figure 1. Resource Metrics Pipeline
+
+The architecture components, from right to left in the figure, consist of the following:
+
+* [cAdvisor](https://github.com/google/cadvisor) - Daemon for collecting, aggregating and exposing Docker container metrics. Note that non-Docker container runtimes that support metrics collection through a [CRI will not require cAdvisor](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-node/cri-container-stats.md).
+* [kubelet](https://kubernetes.io/docs/concepts/overview/components/#kubelet) - Node agent for managing container resources. Resource metrics are accessible using the `/metrics/resource` and `/stats` kubelet API endpoints. 
+* [Summary API](#summary-api-source) - Kubelet API for discovering and retrieving per-node summarized stats available through the `/stats` endpoint. 
+* [metrics-server](#metrics-server) - Cluster addon component that collects and aggregates resource metrics pulled from each kubelet. The K8s API server serves up the Metric API endpoints for use by HPA, VPA and the `kubectl top` command.
+* [Metrics API](#metrics-api) - Kubernetes API supporting access to CPU and memory used for workload autoscaling. Metrics Server is the default implementation provided with many popular K8s distributions, however it can be replaced by alternative adapters based on your preferred monitoring solution.
```

Comment:
_**not checked for technical accuracy**_ and probably OK to save for a follow-up PR.
```suggestion
[cAdvisor](https://github.com/google/cadvisor)
: daemon for collecting, aggregating and exposing container metrics included in Kubelet. 
* [kubelet](/docs/concepts/overview/components/#kubelet)
: Node agent for managing container resources. Resource metrics are accessible using the `/metrics/resource` and `/stats` kubelet API endpoints. 
* [Summary API](#summary-api-source)
: API provided by the kubelet for discovering and retrieving per-node summarized stats available through the `/stats` endpoint. 
* [metrics-server](#metrics-server)
: Cluster addon component that collects and aggregates resource metrics pulled from each kubelet. The API server serves Metrics API for use by HPA, VPA, and by the `kubectl top` command. Metrics Server is a reference implementation of the Metrics API.
* [Metrics API](#metrics-api)
: Kubernetes API supporting access to CPU and memory used for workload autoscaling. To make this work in your cluster, you need an API extension server that provides the Metrics API. 

{{< note >}}
cAdvisor supports reading metrics from cgroups, which works with typical container runtimes on Linux.

If you use a container runtime that uses another resource isolation mechanism, for example virtualization, then that container runtime must support [CRI Container Metrics](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-node/cri-container-stats.md) in order for metrics to be available to the kubelet.
{{< /note >}}
```
Let's be vendor-agnostic about container runtime. I also recognize that we might need to revise this again for v1.24

Reply:
commit suggestion greyed out. will add in separate commit